scene/theme/theme_db.cpp

```diff
@@ -188,6 +189,49 @@ Ref<StyleBox> ThemeDB::get_fallback_stylebox() {
 	return fallback_stylebox;
 }
 
+// Theme item binding.
+
+void ThemeDB::bind_class_item(const StringName &p_class_name, const StringName &p_prop_name, const StringName &p_item_name, ThemeItemSetter p_setter) {
+	ERR_FAIL_COND_MSG(theme_item_binds[p_class_name].has(p_prop_name), vformat("Failed to bind theme item '%s' in class '%s': already bound", p_prop_name, p_class_name));
+
+	ThemeItemBind bind;
+	bind.class_name = p_class_name;
+	bind.item_name = p_item_name;
+	bind.setter = p_setter;
+
+	theme_item_binds[p_class_name][p_prop_name] = bind;
+}
+
+void ThemeDB::bind_class_external_item(const StringName &p_class_name, const StringName &p_prop_name, const StringName &p_item_name, const StringName &p_type_name, ThemeItemSetter p_setter) {
+	ERR_FAIL_COND_MSG(theme_item_binds[p_class_name].has(p_prop_name), vformat("Failed to bind theme item '%s' in class '%s': already bound", p_prop_name, p_class_name));
+
+	ThemeItemBind bind;
+	bind.class_name = p_class_name;
+	bind.item_name = p_item_name;
+	bind.type_name = p_type_name;
+	bind.external = true;
+	bind.setter = p_setter;
+
+	theme_item_binds[p_class_name][p_prop_name] = bind;
+}
+
+void ThemeDB::update_class_instance_items(Node *p_instance) {
+	ERR_FAIL_COND(!p_instance);
+
+	// Use the hierarchy to initialize all inherited theme caches. Setters carry the necessary
+	// context and will set the values appropriately.
+	StringName class_name = p_instance->get_class();
+	while (class_name != StringName()) {
+		if (theme_item_binds.has(class_name)) {
+			for (const KeyValue<StringName, ThemeItemBind> &E : theme_item_binds[class_name]) {
```

Comment:
You can avoid double lookup by using something like this
```suggestion
		HashMap<StringName, HashMap<StringName, ThemeDB::ThemeItemBind>>::Iterator E = theme_item_binds.find(class_name);
		if (E) {
			for (const KeyValue<StringName, ThemeItemBind> &F : E->value) {
```